include/kodlab_mjbots_sdk/common_header.h

```diff
@@ -87,19 +87,28 @@ class ValidatedCache
    * @brief Sets the class data and marks it valid
    * @param data valid data
    */
-  void set(const T &data)
+  virtual void set(const T &data)
   {
     data_ = data;
     valid_ = true;
   }
 
+  /**
+   * @brief Uses operator= to set the class data and marks it valid
+   * @param data valid data
+   */
+  void operator=(const T &data)
+  {
+    set(data);
+  }
+
   /**
    * @brief Accessor for data (<b>does not check validity</b>)
    * @return data
    */
   T get()
   {
-    if (!valid_)
+    if (!valid())
     {
       std::cerr << "[WARN] Returning invalid data." << std::endl;
```

Comment:
Now that we are using this more, will this line terminate the code, or just print out an error. It might also be useful to provide a little more info about which cache it is. For now I believe we could print out T.